### PR TITLE
evidence: fix bug with hashes

### PR DESCRIPTION
**Pull request by cmwaters**
This is a non-breaking solution to #6342 where the hashes remain the same. 

Specifically, I have removed `fastCheck` and modified the `CheckEvidence` function so that the bytes of the evidence are not modified.

I have also added some more tests on top of this

Closes: #6326
Closes: #6312 
Addresses: #5942

### CODE REVIEW

**Review by alexanderbez — APPROVED** [review]

File: types/validator_set.go [hunk 1]
```diff
@@ -1028,7 +1028,9 @@ func (vals *ValidatorSet) ToProto() (*tmproto.ValidatorSet, error) {
 	}
 	vp.Proposer = valProposer
 
-	vp.TotalVotingPower = vals.totalVotingPower
+	// NOTE: Sometimes we use the bytes of the proto form as a hash. This means that we need to
+	// be consistent with cached data
+	vp.TotalVotingPower = 0
```
Comment (cmwaters) [inline types/validator_set.go:1033]: Can you confirm that this doesn't break proto

File: types/validator_set.go [hunk 2]
```diff
@@ -1059,7 +1061,10 @@ func ValidatorSetFromProto(vp *tmproto.ValidatorSet) (*ValidatorSet, error) {
 
 	vals.Proposer = p
 
-	vals.totalVotingPower = vp.GetTotalVotingPower()
+	// NOTE: We can't trust the total voting power given to us by other peers. If someone were to
+	// inject a non-zeo value that wasn't the correct voting power we could assume a wrong total
+	// power. We should look to remove TotalVotingPower from proto or add it in the validators hash.
+	vals.totalVotingPower = 0
```
Comment (cmwaters) [inline types/validator_set.go:1067]: and here...

**Review by cmwaters — COMMENTED** [review]
@marbar3778 can you confirm that this makes sense :)

File: types/validator_set.go (hunk 2 above)
Comment (marbar3778) [reply types/validator_set.go:1067]: On fromProto, wouldn't you want to populate it? you could use your local validatorset power, right?

Comment (marbar3778) [reply types/validator_set.go:1067]: talked in slack, this looks good. It should not be breaking.

**Pull request merged**